operator/pkg/helmreconciler/apply.go

```diff
@@ -178,6 +181,18 @@ func (h *HelmReconciler) ApplyObject(obj *unstructured.Unstructured) error {
 	}
 
 	gvk := obj.GetObjectKind().GroupVersionKind()
+
+	// check minor version only
+	if h.restConfig != nil {
+		k8sVer, err := k8sversion.GetKubernetesVersion(h.restConfig)
```

Comment:
Aren't we calling this on every single object? seems super expensive. can we cache it up somewhere?

Reply:
good point, moved the version check before the apply logic call to avoid calling for each single project